cpp/src/table/device_table.cuh

```diff
@@ -206,6 +206,39 @@ struct elements_are_equal {
 }  // namespace
 
 
+struct equality_comparator {
+
+  equality_comparator(device_table const& lhs, bool nulls_are_equal = false) :
+                            _lhs(lhs), _rhs(lhs), _nulls_are_equal(nulls_are_equal) {
+  }
+  equality_comparator(device_table const& lhs, device_table const& rhs, 
+                                                  bool nulls_are_equal = false) :
+                            _lhs(lhs), _rhs(rhs), _nulls_are_equal(nulls_are_equal) {
+  }
+
+  __device__ inline bool operator()(gdf_index_type lhs_index, gdf_index_type rhs_index) {
```

Comment:
This seems to be doing much the same thing as `rows_equal` below. What's the intended difference?

Reply:
I wanted an binary operator that I could initialize and pass into a thrust function. `rows_equal` is a function with five parameters. I know they are almost the same and I struggled for a while about how to do both without being too redundant. I am open to suggestions.

Reply:
Making the functor is fine, the body of the `operator()` should just call `rows_equal` to reduce duplicated code.

Reply:
I don't know why I thought that would be inefficient.... but yeah, that is the obvious thing to do. DONE!